### ISSUE TITLE
MSYS2-UCRT ICE and segfault on 

### ISSUE THREAD

**Issue by willwray** [cschreib/snitch#103]
My wrapDB snitch PR CI is all passing apart from [MSYS2-UCRT64](https://github.com/mesonbuild/wrapdb/actions/runs/5105901488/jobs/9177983445?pr=1036)

```c++
Compiling C++ object subprojects/snitch-1.1.1/libsnitch.a.p/src_snitch.cpp.obj
FAILED: subprojects/snitch-1.1.1/libsnitch.a.p/src_snitch.cpp.obj 
"c++" "-Isubprojects/snitch-1.1.1/libsnitch.a.p" "-Isubprojects/snitch-1.1.1" "-I../subprojects/snitch-1.1.1" "-I../subprojects/snitch-1.1.1/include" "-fdiagnostics-color=always" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-std=c++20" "-O0" "-g" -MD -MQ subprojects/snitch-1.1.1/libsnitch.a.p/src_snitch.cpp.obj -MF "subprojects/snitch-1.1.1/libsnitch.a.p/src_snitch.cpp.obj.d" -o subprojects/snitch-1.1.1/libsnitch.a.p/src_snitch.cpp.obj "-c" ../subprojects/snitch-1.1.1/src/snitch.cpp
In file included from ../subprojects/snitch-1.1.1/include/snitch/snitch_string.hpp:5,
                 from ../subprojects/snitch-1.1.1/include/snitch/snitch_append.hpp:7,
                 from ../subprojects/snitch-1.1.1/src/snitch_append.cpp:1,
                 from ../subprojects/snitch-1.1.1/src/snitch.cpp:1:
../subprojects/snitch-1.1.1/include/snitch/snitch_vector.hpp:215:37: internal compiler error: Segmentation fault
  215 |     std::array<ElemType, MaxLength> data_buffer = {};
      |                                     ^~~~~~~~~~~
```

It may be some environment or resource issue with that particular CI build.

Opening as an issue here in case it smokes out a problem in snitch code.

**Comment by tocic** [comment]
That's a gcc bug — https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109730.

**Comment by cschreib** [comment]
Thank you for reporting this issue, and thanks @tocic for reporting the bug back to GCC. Looks like a fix has been found and will be released in GCC 12.4, 13.2, 14.0. I assume GCC <= 11 isn't affected. Closing as "not our bug".
